akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -124,37 +130,40 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
             snapshotWhen,
             recovery,
             retention,
+            journalResponses,
+            snapshotResponses,
+            permitterResponses,
             holdingRecoveryPermit = false,
             settings = settings,
             stashState = stashState)
 
-          // needs to accept Any since we also can get messages from the journal
-          // not part of the protocol
-          val onStopInterceptor = new BehaviorInterceptor[Any, Any] {
-
+          // needs to accept Any since we can receive both Command and InternalProtocol
+          val interceptor = new BehaviorInterceptor[Any, InternalProtocol] {
+            // FIXME is replacement of this interceptor working for restarts, or do we need isSame impl?
             import BehaviorInterceptor._
-            def aroundReceive(ctx: typed.TypedActorContext[Any], msg: Any, target: ReceiveTarget[Any])
-                : Behavior[Any] = { target(ctx, msg) }
 
-            def aroundSignal(ctx: typed.TypedActorContext[Any], signal: Signal, target: SignalTarget[Any])
-                : Behavior[Any] = {
+            def aroundReceive(ctx: typed.TypedActorContext[Any], msg: Any, target: ReceiveTarget[InternalProtocol])
+                : Behavior[InternalProtocol] = {
+              msg match {
+                case internal: InternalProtocol => target(ctx, internal) // such as RecoveryTickEvent
+                case cmd: Command @unchecked    => target(ctx, InternalProtocol.IncomingCommand(cmd))
```

Comment:
This interceptor could convert the JournalProtocol.Response, SnapshotProtocol.Response and RecoveryPermitter.RecoveryPermitGranted instead of using the message adapters.